CMakeLists.txt

```diff
@@ -20,6 +20,8 @@ mbed_set_cpu_core_options(mbed-os ${MBED_TOOLCHAIN})
 
 include(${MBED_ROOT}/cmake/toolchain.cmake)
 mbed_set_toolchain_options(mbed-os)
+mbed_set_language_standard(${APP_TARGET})
```

Comment:
Should this call be made in the application instead?

I could have set the variable `CMAKE_<LANG>_STANDARD` (instead of the target property as done in `toolchain.cmake`) in Mbed OS but it would have only applied to Mbed OS.

My understanding of the documentation is that when the variable is set it only applies to the targets defined after that. Therefore application targets must set their properties instead to select the language standard.

Reply:
it seems this won't be the first neither last app specific setting ? We shall move out of it here and maybe providing a function that sets the default (Mbed OS default - examples would use it).

Reply:
Yes, it may well not be the only thing that is app target specific. So yes it would be good to have one function that the app can call and pass its target name rather than calling multiple functions.
So we will do that later then?

Reply:
@0xc0170 
Actually the build fails because the call needs to be made from where the target has been created.
The CI is failing because of that. So it has to be move the app. Are we ok with that?

Reply:
if that is the case, would it make sense to provide a default function apps can use (not copy-paste 5 functions to call but rather use one to set the toolchain configuration) ?  Or shouldn't this be teh default behavior and we provide a way to override it  (not certain how, using -Dsomething does not look that appealing)

Reply:
Done:
* https://github.com/ARMmbed/mbed-os/compare/82076cf967c6bd11a7336153a8d55b05a849ec1f..a8e482b680a37799737c419c9225c4731b58eff0
* https://github.com/ARMmbed/mbed-os-example-blinky/pull/227